opentracing-jaxrs2/src/main/java/io/opentracing/contrib/jaxrs2/server/ServerTracingFilter.java

```diff
@@ -59,7 +57,8 @@ public void filter(ContainerRequestContext requestContext) throws IOException {
                 spanBuilder.asChildOf(extractedSpanContext);
             }
 
-            Span span = spanBuilder.start();
+            Span span = spanBuilder.startManual();
+            tracer.makeActive(span);
```

Comment:
One possible approach would be to replace in the request filter `requestContext.setProperty(SPAN_PROP_ID, new SpanWrapper(span));` by a request context property that stores the `Continuation` returned from `span.capture()` - and on a following line do `span.deactivate()`.

Then in the response filter retrieve the `Continuation` and `activate` it to retrieve the active span which can then be `deactivate`d at the end of the method.